nixopsaws/resources/ec2_keypair.py

```diff
@@ -35,104 +45,94 @@ class EC2KeyPairState(nixops.resources.ResourceState):
     keypair_name = nixops.util.attr_property("ec2.keyPairName", None)
     public_key = nixops.util.attr_property("publicKey", None)
     private_key = nixops.util.attr_property("privateKey", None)
+    profile = nixops.util.attr_property("ec2.profile", None)
     access_key_id = nixops.util.attr_property("ec2.accessKeyId", None)
     region = nixops.util.attr_property("ec2.region", None)
 
-
     @classmethod
     def get_type(cls):
         return "ec2-keypair"
 
-
     def __init__(self, depl, name, id):
-        nixops.resources.ResourceState.__init__(self, depl, name, id)
-        self._conn = None
-
+        super(EC2KeyPairState, self).__init__(depl, name, id)
+        self._session = None  # type: boto3.session.Session
 
     def show_type(self):
         s = super(EC2KeyPairState, self).show_type()
-        if self.region: s = "{0} [{1}]".format(s, self.region)
-        return s
+        if self.region:
+            s = "{0} [{1}]".format(s, self.region)
 
+        return s
```

Comment:
I'd consider upstreaming all of these changes and some of the mypy annotations in a separate no-op cleanup PR, to make it easier to review the functional changes.  

It'd be nice if we just hose down the whole project with an auto-formatter like `yapf` or `black` and call it a day.  I don't have any real style preference personally, but those tools do a pretty decent job and free us up to look at more interesting things.

Reply:
We use `black` now in NixOS tests as well.